aws/resource_aws_wafv2_web_acl_association_test.go

```diff
@@ -0,0 +1,142 @@
+package aws
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/wafv2"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+)
+
+func TestAccAwsWafv2WebACLAssociation_Basic(t *testing.T) {
+	testName := fmt.Sprintf("web-acl-association-%s", acctest.RandString(5))
+	resourceName := "aws_wafv2_web_acl_association.test"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSWafv2WebACLAssociationDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAwsWafv2WebACLAssociationConfig(testName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSWafv2WebACLAssociationExists(resourceName),
```

Comment:
add tests for arn using the test func, i.e. `testAccMatchResourceAttrRegionalARN()`